cadc-access-control-identity/src/main/java/ca/nrc/cadc/auth/ACIdentityManager.java

```diff
@@ -247,6 +248,20 @@ public Subject toSubject(Object o) {
     }
 
     public void augmentSubject(final Subject subject) {
+
+        if (subject == null ) {
+            return;
+        }
+
+        // If the principal list is in the subject has  Numeric Principal
+        // AND the list is greater than 1, then LDAP doesn't need to be
+        // called here
+        Set<Principal> principalSet = subject.getPrincipals();
+        Set<NumericPrincipal> nPrincipalSet = subject.getPrincipals(NumericPrincipal.class);
+        if (principalSet != null && principalSet.size() > 1 && nPrincipalSet != null ) {
```

Comment:
The API to subject.getPrincipals says that it creates a new Set each invocation, so this line can be as simple as if (!principalSet.isEmpty())